### PR TITLE
Adding improv_serial to configs for web-based installation firmware.

### PR DESCRIPTION
**Pull request by barndawgie**
I've tested this via https://barndawgie.github.io/install/

There are two main caveats with this:

1. In theory, improv_serial should immediately connect after install to allow wi-fi configuration. In practice this fails and you need to hit "Connect" a second time to get the Configure Wi-Fi option. I believe this needs to be filed as an ESP Web Tools bug - will do that once we get this finalized and tested by a few more people.
2. improv relies on UART0 for communication, so esphome-econet needs to move to UART1. This is fine for esp32 devices, but could be a problem (and requires testing) with ESP8266 devices since this implies moving to software UART for communication with the Rheem appliance.

### CODE REVIEW

**Comment by barndawgie** [comment]
Converting this to a draft until I can get someone to test on an ESP8266 device.

File: build-yaml/esp32.yaml [hunk 1]
```diff
@@ -4,6 +4,15 @@ substitutions:
   rx_pin: GPIO22
   platform: esp32
   board: esp32dev
+  uart_id: uart_1
 
 wifi:
   ap:
+
+logger:
+  baud_rate: 115200  # Make sure logging is not using the serial port
```
Comment (tronikos) [inline build-yaml/esp32.yaml:13]: Remove comment

File: build-yaml/esp8266.yaml [hunk 2]
```diff
@@ -4,6 +4,15 @@ substitutions:
   rx_pin: GPIO5
   platform: esp8266
   board: d1_mini
+  uart_id: uart_1
 
 wifi:
   ap:
+
+logger:
+  baud_rate: 115200  # Make sure logging is not using the serial port
```
Comment (tronikos) [inline build-yaml/esp8266.yaml:13]: Remove comment

**Comment by tronikos** [comment]
I tried on ESP8266. Everything seems to work except:

1. logs

2. Adapting the device in the ESPHome dashboard the substitutions were lost so it compiled for ESP32 and used the wrong pins. I had to add:
```
  tx_pin: GPIO4
  rx_pin: GPIO5
  platform: esp8266
  board: d1_mini
```

3. When installing I was presented with:
"Confirm Installation
Do you want to install esphome-econet improv-serial?

All data on the device will be erased."

In a follow up PR can you rename the device to match the selected device?

**Review by tronikos — APPROVED** [review]

**Pull request merged**